src/utils/StorageHelper.ts

```diff
@@ -0,0 +1,83 @@
+import { getPublic, sign } from "@toruslabs/eccrypto";
+import { get, post } from "@toruslabs/http-helpers";
+import { decryptData, encryptData, keccak256 } from "@toruslabs/metadata-helpers";
+
+import { LoginDetails } from "../handlers/interfaces";
+import { REDIRECT_PARAMS_STORAGE_METHOD, REDIRECT_PARAMS_STORAGE_METHOD_TYPE } from "./enums";
+import { are3PCSupported, storageAvailable } from "./helpers";
+
+class StorageHelper {
+  private currentStorageMethod: REDIRECT_PARAMS_STORAGE_METHOD_TYPE = REDIRECT_PARAMS_STORAGE_METHOD.LOCAL_STORAGE;
+
+  private isInitialized = false;
+
+  private storageServerUrl = "https://broadcast-server.tor.us";
+
+  constructor(serverUrl: string) {
+    this.storageServerUrl = serverUrl;
+  }
+
+  init() {
+    const support = are3PCSupported();
+    const localStorageAvailable = storageAvailable(REDIRECT_PARAMS_STORAGE_METHOD.LOCAL_STORAGE);
+    if (support && localStorageAvailable) {
+      // use local storage as default for storing stuff
+      this.currentStorageMethod = REDIRECT_PARAMS_STORAGE_METHOD.LOCAL_STORAGE;
+    } else {
+      // use server store as default for storing stuff
+      this.currentStorageMethod = REDIRECT_PARAMS_STORAGE_METHOD.SERVER;
+    }
+    this.isInitialized = true;
+  }
+
+  async storeLoginDetails(params: LoginDetails, scope: string): Promise<void> {
+    if (!this.isInitialized) throw new Error("StorageHelper is not initialized");
+    if (this.currentStorageMethod === REDIRECT_PARAMS_STORAGE_METHOD.SERVER) {
+      const privKey = keccak256(scope);
+      const privKeyHex = privKey.toString("hex");
+      const publicKeyHex = getPublic(privKey).toString("hex");
+      const encData = await encryptData(privKeyHex, params);
```

Comment:
encryption should be done using public key.

Reply:
the method internally derives the public key and uses it to encrypt.
it's just a helper function